File: 1-two-sum/1-two-sum.cpp
  class Solution {
  public:
      vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> umap;
        for (int index = 0; index < nums.size(); index++) {
            int y = target - nums[index];
            if (umap.find(y) != umap.end()) {
                return {umap[y], index};
            }
            umap[nums[index]] = index;
        }
        return {};
    }
  };

File: 1-two-sum/1-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  if (nums.length === 0) return [-1, -1];

  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    if (map.has(nums[i])) {
      return [map.get(nums[i]), i];
    } else {
      map.set(target - nums[i], i);
    }
  }

  return [-1, -1];
}